src/main/java/com/uber/cadence/converter/JsonDataConverter.java

```diff
@@ -165,7 +166,12 @@ public JsonDataConverter(Function<GsonBuilder, GsonBuilder> builderInterceptor)
       JsonArray array = parser.parse(new String(content, StandardCharsets.UTF_8)).getAsJsonArray();
       Object[] result = new Object[valueTypes.length];
       for (int i = 0; i < valueTypes.length; i++) {
-        result[i] = gson.fromJson(array.get(i), valueTypes[i]);
+
+        if(i >= array.size()){ //Missing arugments => add defaults
+          result[i] = Defaults.defaultValue(valueTypes[i].getClass());
```

Comment:
valueTypes[i].getClass is always going to return java.lang.reflect.Type class.
I think you want to cast Type to Class and if not possible use null as a default.